pkg/kn/commands/namespaced_test.go

```diff
@@ -143,71 +143,116 @@ func TestGetNamespaceFallback(t *testing.T) {
+		assert.NilError(t, err)
 
-	// Invalid kubeconfig
-	kp := &KnParams{KubeCfgPath: tempFile}
-	actual, err := kp.CurrentNamespace()
-	assert.Assert(t, err != nil)
-	assert.Assert(t, clientcmd.IsConfigurationInvalid(err))
-	assert.Assert(t, actual == "")
-
-	// Missing kubeconfig
-	kp = &KnParams{KubeCfgPath: filepath.Join(tempDir, "missing")}
-	actual, err = kp.CurrentNamespace()
-	assert.Assert(t, err != nil)
-	assert.ErrorContains(t, err, "can not be found")
-	assert.Assert(t, actual == "")
-
-	// Variable namespace override
-	kp = &KnParams{fixedCurrentNamespace: FakeNamespace}
-	actual, err = kp.CurrentNamespace()
-	assert.NilError(t, err)
-	assert.Equal(t, actual, FakeNamespace)
+		kp := &KnParams{KubeCfgPath: tempFile}
+		actual, err := kp.CurrentNamespace()
+		if isInCluster() {
+			// In-cluster config overrides the mocked one in OpenShift CI
+			assert.NilError(t, err)
+			assert.Equal(t, actual, os.Getenv("NAMESPACE"))
+		} else {
+			assert.Assert(t, err != nil)
+			assert.Assert(t, clientcmd.IsConfigurationInvalid(err))
+			assert.Assert(t, actual == "")
+		}
 
-	// Fallback to "default" namespace from mock kubeconfig
-	tempFile = filepath.Join(tempDir, "mock")
-	err = ioutil.WriteFile(tempFile, []byte(BASIC_KUBECONFIG), test.FileModeReadWrite)
-	assert.NilError(t, err)
-	kp = &KnParams{KubeCfgPath: tempFile}
-	actual, err = kp.CurrentNamespace()
-	assert.NilError(t, err)
-	assert.Equal(t, actual, "default")
+	})
+
+	t.Run("MissingConfig", func(t *testing.T) {
+		// Missing kubeconfig
+		kp := &KnParams{KubeCfgPath: filepath.Join(tempDir, "missing")}
+		actual, err := kp.CurrentNamespace()
+		assert.Assert(t, err != nil)
+		assert.ErrorContains(t, err, "can not be found")
+		assert.Assert(t, actual == "")
+	})
+
+	t.Run("MissingConfig", func(t *testing.T) {
+		// Variable namespace override
+		kp := &KnParams{fixedCurrentNamespace: FakeNamespace}
+		actual, err := kp.CurrentNamespace()
+		assert.NilError(t, err)
+		assert.Equal(t, actual, FakeNamespace)
+	})
+
+	t.Run("MockConfig", func(t *testing.T) {
+		// Fallback to "default" namespace from mock kubeconfig
+		tempFile := filepath.Join(tempDir, "mock")
+		err = ioutil.WriteFile(tempFile, []byte(BASIC_KUBECONFIG), test.FileModeReadWrite)
+		assert.NilError(t, err)
+		kp := &KnParams{KubeCfgPath: tempFile}
+		actual, err := kp.CurrentNamespace()
+		assert.NilError(t, err)
+		if isInCluster() {
+			// In-cluster config overrides the mocked one in OpenShift CI
+			assert.Equal(t, actual, os.Getenv("NAMESPACE"))
+		} else {
+			assert.Equal(t, actual, "default")
+		}
+	})
+}
+
+// Inspired by client-go function
+// https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L600-L606
+func isInCluster() bool {
+	fi, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	return os.Getenv("KUBERNETES_SERVICE_HOST") != "" &&
+		os.Getenv("KUBERNETES_SERVICE_PORT") != "" &&
+		err == nil && !fi.IsDir()
```

Comment:
when can `fi` be a directory ? (academic question ;-)

Looks good to me.

/approve
/lgtm